cmd/dist/edit.go

```diff
@@ -109,5 +110,25 @@ func edit(rd io.Reader) (io.ReadCloser, error) {
 		return nil, err
 	}
 
-	return tmp, nil
+	return onCloser{tmp, onClose: func() error {
+		return os.RemoveAll(tmp.Name())
+	}}, nil
+}
+
+type onCloser struct {
+	io.ReadCloser
+	onClose func() error
+}
+
+func (oc onCloser) Close() error {
+	var err error
+	if oc.onClose != nil {
```

Comment:
> do you think we should still change the order to close the resource before deleting it?

I am really not sure. This could have unexpected behavior for non-posix filesystems (anything without inodes).